src/connectors/infinite-hits/connectInfiniteHits.js

```diff
@@ -130,6 +147,22 @@ export default function connectInfiniteHits(renderFn, unmountFn) {
       dispose() {
         unmountFn();
       },
+
+      getWidgetState(uiState, { searchParameters }) {
+        const page = searchParameters.page;
+        if (page === 0 || page + 1 === uiState.page) return uiState;
+        return {
+          ...uiState,
+          page: page + 1,
+        };
+      },
+
+      getWidgetSearchParameters(searchParameters, { uiState }) {
+        const uiPage = uiState.page;
+        if (uiPage)
+          return searchParameters.setQueryParameter('page', uiState.page - 1);
```

Comment:
The variable `uiPage` is declared but not used here.

```suggestion
          return searchParameters.setQueryParameter('page', uiPage - 1);
```